bin/tests/it/k8s.rs

```diff
@@ -200,6 +200,92 @@ async fn delete_service(client: Client, service_name: &str, namespace: &str) {
     }
 }
 
+async fn delete_daemonset(client: Client, daemonset_name: &str, namespace: &str) {
```

Comment:
I think we can do a better job to ensure RAII with Rust's language features. Maybe we instead have the `create_agent_ds` function return a `AgentGuard` which is comparable to a `MutexGuard` which we get after locking a mutex - this resource can implement these delete operations with the drop trait

Reply:
same as before - this is test code, explicit order etc, easy to debug. the Drop trait's drop method isn't asynchronous. so have to spawn new task. and then deal with borrow/ownership rules of cleanup params. complexity is increasing and new code will need to be tested itself.